### PR TITLE
mrc-2379 Show read-only view of parameters

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This branch adds the sidebar with read-only parameters, using the new readonly mode in vue-dynamic-form. 

The version of Vue used has been reverted to Vue 2.6 in order to use vue-dynamic-form and its dependent packages, with CompositionAPI included to allow us to continue to use that syntax in comet components

A new Parameters component receives Parameters metadata, and creates a dynamic form for each parameter group of type 'dynamicForm' (the other type, 'rt' - phases - to be implemented in another ticket). 

To display the readonly sections, the component adds 'collapsible' and 'collapsed' fields to the metadata - these are not included in the metadata sent from the server because we won't want to collapse the editable forms, to be shown in separate dialogs. 

The action which fetches metadata now also rehydrates the parameters jsonata into json before committing. We do this here rather than in the component (as is done for Charts metadata) because I think it will be preferable to keep both the latest form metadata (including updated values) and data (values only, to be sent to the backend) together in the store state, rather than internally managed by the Parameters component - though we could do it either way.

### CODE REVIEW

File: src/app/static/src/views/Home.vue [hunk 1]
```diff
@@ -33,6 +38,7 @@ export default defineComponent({
         ])
     },
     mounted() {
+        console.log("mounted");
```
Comment (mwoodbri) [inline src/app/static/src/views/Home.vue]: No harm in this... but deliberate?

**Review by mwoodbri — CHANGES_REQUESTED** [review]
Appreciate this has involved a lot of faff with Vue etc. But disregarding that I'm impressed how small this diff is, and the end result looks great!

I'm agnostic about where we rehydrate the jsonata, and shouldn't be too hard to change if we need to. On a related note... I can already see some pain from jsonata not being json, and therefore the lack of help that editor etc can provide. No easy solution for that though.

Only a couple of cosmetic issues that we might consider addressing, now or later. The accordion starts open and then immediately collapses for me - is this intended? And the spacing in form is less than ideal - if it was more full-width then the labels would be easier to read:

comet_sidebar

**Comment by EmmaLRussell** [comment]
> Only a couple of cosmetic issues that we might consider addressing, now or later. The accordion starts open and then immediately collapses for me - is this intended? And the spacing in form is less than ideal - if it was more full-width then the labels would be easier to read:
> 

The collapsed sections starting off opened seems to be a limitation of the form - it happens on the example form too, so I'll see if I can fix it there. I'll see what I can do about the label wrapping here - the form does apply col classes to labels and values, but its other clients expect this.

**Review by mwoodbri — APPROVED** [review]
Sidebar looking great now

**Review by LekanAnni — APPROVED** [review]

**Pull request merged**